appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -79,6 +79,9 @@ export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
     }
 
     public async redeployDeployment(): Promise<void> {
+        if (this._deployResult.isReadonly) {
+            throw new Error(localize('redeployNotSupported', 'Redeploy does not support non-git deployments.'));
```

Comment:
nit: I like your previous wording better. Specifically, `Redeploy is not supported for` instead of `Redeploy does not support`

Reply:
No strong preference